### PR TITLE
Fix data_source_azure_access_credentials US Government Cloud

### LINKED ISSUE #1589: data_source_azure_access_credentials Does not support US Government Cloud
The `vault_azure_access_credentials` data source currently does not function with US Government cloud.

Per the documentation the required environment input is: `AzureGovernmentCloud`

However using this value results in an error:
`Error: autorest/azure: There is no cloud environment matching the name "AZUREGOVERNMENTCLOUD"`

A quick dig reveals a mismatch between the `go-autorest/azure/` repo and `azure-sdk-for-go/arm`

---

Go Autorest, expects `AzureUSGovernmentCloud`. The call to this sdk is here:
https://github.com/hashicorp/terraform-provider-vault/blob/c3dcbc3bddccbff2f34352b9c11fca4413e96619/vault/data_source_azure_access_credentials.go#L227

SDK Reference:
https://github.com/Azure/go-autorest/blob/2fa44cb18b8338d7fa4f749bb798d6cbb3d9ba0c/autorest/azure/environments.go#L40

Azure SDK For GO Expects ``. Provider call:
https://github.com/hashicorp/terraform-provider-vault/blob/c3dcbc3bddccbff2f34352b9c11fca4413e96619/vault/data_source_azure_access_credentials.go#L237

SDK Reference:
https://github.com/Azure/azure-sdk-for-go/blob/17ab49c9c762feefa85d9362a13d278bdea41f43/sdk/azcore/arm/arm.go#L16

### PR DESCRIPTION
**Pull request by jhisc**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #1589

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
fix: data_source_azure_access_credentials environment support for government cloud
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -run=TestAccXXX -timeout 30m ./...
?       github.com/hashicorp/terraform-provider-vault   [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/coverage      [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/generate      [no test files]
ok      github.com/hashicorp/terraform-provider-vault/codegen   0.067s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/generated [no test files]
ok      github.com/hashicorp/terraform-provider-vault/generated/datasources/transform/decode    0.039s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/generated/datasources/transform/encode    0.089s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/alphabet    0.039s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/role        0.056s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/template    0.035s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/transformation      0.031s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/helper    [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/consts   [no test files]
ok      github.com/hashicorp/terraform-provider-vault/internal/identity/entity  0.034s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/internal/identity/group   [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/pki      [no test files]
ok      github.com/hashicorp/terraform-provider-vault/internal/provider 0.057s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/internal/semver   0.032s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/schema    [no test files]
ok      github.com/hashicorp/terraform-provider-vault/testutil  0.031s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/util      0.030s [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/vault     0.747s [no tests to run]
```

### CODE REVIEW

**Comment by jhisc** [comment]
This PR aligns the inputs with the autorest SDK:
https://github.com/Azure/go-autorest/blob/2fa44cb18b8338d7fa4f749bb798d6cbb3d9ba0c/autorest/azure/environments.go#L34

File: vault/data_source_azure_access_credentials.go [hunk 1]
```diff
@@ -234,7 +234,7 @@ func azureAccessCredentialsDataSourceRead(d *schema.ResourceData, meta interface
 			clientOptions.Endpoint = arm.AzurePublicCloud
 		case "AzureChinaCloud":
 			clientOptions.Endpoint = arm.AzureChina
-		case "AzureGovernmentCloud":
+		case "AzureUSGovernmentCloud":
```
Comment (benashz) [inline vault/data_source_azure_access_credentials.go:237]: I believe this change will break anyone who has configured `AzureGovernmentCloud`, suggest adding `AzureGovernmentCloud` :

```suggestion
		case "AzureGovernmentCloud", "AzureUSGovernmentCloud":
```

Comment (jhisc) [reply vault/data_source_azure_access_credentials.go:237]: @benashz - Thank you for reviewing.

Anyone that is currently using `AzureGovernmentCloud` would receive an error from the autorest SDK:
`Error: autorest/azure: There is no cloud environment matching the name "AZUREGOVERNMENTCLOUD"`

Happy to add it if you want, but the current value is not supported and cannot be used without terraform throwing an error. (Referenced in the issue)

Comment (benashz) [reply vault/data_source_azure_access_credentials.go:237]: Ah, yes I see, we should receive an error when getting the environment by name in this case. In that case I think we can skip this suggestion.

**Review by benashz — APPROVED** [review]
LGTM!

Thank you for your contribution to HashiCorp!

**Pull request merged**